### PR TITLE
doc/rados/ops: edit health-checks.rst (3 of x)

### PR DESCRIPTION
**Pull request by zdover23**
Edit docs/rados/operations/health-checks.rst (3 of x).

Follows https://github.com/ceph/ceph/pull/50825

https://tracker.ceph.com/issues/58485

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: doc/rados/operations/health-checks.rst [hunk 1]
```diff
@@ -539,32 +539,33 @@ or delete some existing data to reduce utilization.
...
-`db` partitions (storage space for metadata, normally on a faster
-device) but that space has filled, such that metadata has "spilled
-over" onto the normal slow device.  This isn't necessarily an error
-condition or even unexpected, but if the administrator's expectation
-was that all metadata would fit on the faster device, it indicates
-that not enough space was provided.
+One or more OSDs that use the BlueStore back end have been allocated `db`
+partitions (that is, storage space for metadata, normally on a faster device),
+but because that space has been filled, metadata has "spilled over" onto the
+normal slow device. This is not necessarily an error condition or even
```
Comment (anthonyeleven) [inline doc/rados/operations/health-checks.rst:545]: s/normal/

File: doc/rados/operations/health-checks.rst [hunk 2]
```diff
@@ -539,32 +539,33 @@ or delete some existing data to reduce utilization.
...
+unexpected behavior, but if the administrator had expected that all metadata
```
Comment (anthonyeleven) [inline doc/rados/operations/health-checks.rst:546]: unexpected behavior but may result in performance degradation.  If the administrator expects that all metadata iwll fit

File: doc/rados/operations/health-checks.rst [hunk 3]
```diff
@@ -663,23 +671,25 @@ This warning can be disabled with:
...
-Starting with the Octopus release, BlueStore tracks omap space utilization
-by pool, and one or more OSDs have volumes that were created prior to
-Octopus.  If all OSDs are not running BlueStore with the new tracking
-enabled, the cluster will report and approximate value for per-pool omap usage
-based on the most recent deep-scrub.
+One or more OSDs have volumes that were created prior to the Octopus release.
+(In Octobus and later releases, BlueStore tracks omap space utilization by
+pool.)
+
+If there are any OSDs that are not running BlueStore with the new tracking
```
Comment (anthonyeleven) [inline doc/rados/operations/health-checks.rst:678]: If there are any BlueStore OSDs that do not have the new tracking

File: doc/rados/operations/health-checks.rst [hunk 4]
```diff
@@ -663,23 +671,25 @@ This warning can be disabled with:
...
+enabled, the cluster will report an approximate value for per-pool omap usage
+based on the most recent deep scrub.
 
-The old OSDs can be updated to track by pool by stopping each OSD,
-running a repair operation, and the restarting it.  For example, if
-``osd.123`` needed to be updated,:
+The old OSDs can be updated to track by pool by stopping each OSD, running a
```
Comment (anthonyeleven) [inline doc/rados/operations/health-checks.rst:682]: s/old/

File: doc/rados/operations/health-checks.rst [hunk 5]
```diff
@@ -712,13 +724,14 @@ This warning can be disabled with:
...
+the physical device and the metadata that tracks its size. This inconsistency
+can lead to the OSD(s) crashing in the future.
 
-The OSDs in question should be destroyed and reprovisioned.  Care should be
-taken to do this one OSD at a time, and in a way that doesn't put any data at
-risk.  For example, if osd ``$N`` has the error:
+The OSDs that have this inconsistency should be destroyed and reprovisioned. Be
+very careful to execute this procedure on only one OSD at a time, so as to
+minimize the risk of losing any data. To execute this procedure, where ``$N``
+is the OSD that has the inconsistency, run the following commands:
```
Comment (anthonyeleven) [inline doc/rados/operations/health-checks.rst:734]: Maybe add a note about waiting for recovery to complete before proceeding to the next.

File: doc/rados/operations/health-checks.rst [hunk 6]
```diff
@@ -731,44 +744,42 @@ risk.  For example, if osd ``$N`` has the error:
...
 ______________________________
 
-One or more OSDs using BlueStore detects spurious read errors at main device.
-BlueStore has recovered from these errors by retrying disk reads.
-Though this might show some issues with underlying hardware, I/O subsystem,
-etc.
-Which theoretically might cause permanent data corruption.
-Some observations on the root cause can be found at 
-https://tracker.ceph.com/issues/22464
+One or more BlueStore OSDs detect spurious read errors on the main device.
```
Comment (anthonyeleven) [inline doc/rados/operations/health-checks.rst:760]: Seems like an odd name, perhaps _transient_ would be better but it's baked in now.

**Review by anthonyeleven — APPROVED** [review]

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/50952
https://github.com/ceph/ceph/pull/50952

**Pull request merged**